# nafld-exercise-dge
This project contains the code needed to conduct processing and preliminary analysis of a RNA-seq dataset that aims to aid in studying the effects of exercise on mice induced with non-alcoholic fatty liver disease (NAFLD).

## Repository Organization

### `scripts/`
Contains all the bash scripts and R scripts that are needed to run the data processing steps. These scripts should be run in the following order:

1. `01_download_samples.sh` downloads all the fastq files associated with the dataset
2. `02_run_fastqc` runs the FastQC analysis and MultiQC aggregation of the downloaded fastq files
3. `03_trim_samples` trims the fastq files using the MGI adapters specified in the GEO project description. FastQC and MultiQC of the trimmed samples is also conducted in this script
4. `04_generate_reference.sh` generates the GRCm39 STAR reference index
5. `05_align_samples.sh` aligns the trimmed fastq files using STAR
6. `06_run_qorts` runs the QORTS post-alignment quality control metrics, generated compiled reports, and runs MultiQC on the QORTS outputs
7. `07_count_reads.sh` runs FeatureCounts on the aligned samples, summarizing on the gene-level
8. `process_DESeq2_object.R` reads the FeatureCounts output into a DESeq2 object and performs normalization and variance stabilization

### `data/`
Contains any data generated from the differential gene expression analysis that may be useful for further analysis

* `deseq2_object_workspace.RData` is a data file that can be loaded into R which contains all of the workspace variables that were generated by running the `process_DESeq2_object.R` script
* `AE_vs_HFD_DGE.csv` contains the gene id, log 2 fold change, p-value, and adjusted p-value for the AE vs HFD DGE
* `HFD_vs_NC_DGE.csv` contains the gene id, log 2 fold change, p-value, and adjusted p-value for the HFD vs NC DGE
* `RE_vs_HFD_DGE.csv` contains the gene id, log 2 fold change, p-value, and adjusted p-value for the RE vs HFD DGE

### `reports/`
Contains the RMarkdown scripts that are used to conduct the analysis and report the results

* `read_counts_exploration.Rmd` loads in data generated by the `process_DESeq2_object.R` script and performs exploratory data analysis on the read counts and samples
* `DGE_GSEA.Rmd` loads in data generated by the `process_DESeq2_object.R` script and performs differential gene expression analysis and GSEA for various condition comparisons
* `final_report.Rmd` contains an overview of the entire project, detailing the methods, results, and discussion

### `qc/`
Contains any high level reports (usually just the MultiQC reports and not individual reports) that were generated from the bash scripts as a part of the quality control analysis. QORTS data on the gene body coverage is also included here, which is referenced to generate a plot in the final report.

### `images/`
Contains any images from the MultiQC reports that are used in the final report



